File: event_hub_back/lib/models/notification.dart
class AppNotification {
  final String notifId;
  final String userId;
  final String titre;
  final String corps;
  final String type;
  final Map<String, String> data;
  final bool lue;
  final DateTime envoyeAt;

  AppNotification({
    required this.notifId,
    required this.userId,
    required this.titre,
    required this.corps,
    required this.type,
    this.data = const {},
    this.lue = false,
    required this.envoyeAt,
  });

  bool get isRead => lue;
  bool get isBookingConfirmed => type == 'booking_confirmed';
  bool get isEventReminder => type == 'event_reminder';
  bool get isTicketReady => type == 'ticket_ready';
  bool get isCancellation => type == 'cancellation';
  bool get isPromotion => type == 'promotion';

  factory AppNotification.fromMap(Map<String, dynamic> map) {
    return AppNotification(
      notifId: map['notifId'] ?? map['id'] ?? '',
      userId: map['userId'] ?? '',
      titre: map['titre'] ?? '',
      corps: map['corps'] ?? '',
      type: map['type'] ?? '',
      data: Map<String, String>.from(map['data'] ?? {}),
      lue: map['lue'] ?? false,
      envoyeAt: map['envoyeAt']?.toDate() ?? DateTime.now(),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'notifId': notifId,
      'userId': userId,
      'titre': titre,
      'corps': corps,
      'type': type,
      'data': data,
      'lue': lue,
      'envoyeAt': envoyeAt,
    };
  }
}

File: event_hub_spring/src/main/java/com/eventhub/entity/Notification.java
package com.eventhub.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@Entity
@Table(name = "notifications")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "notif_id", updatable = false, nullable = false)
    private String notifId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_uid", nullable = false)
    private User user;

    @Column(nullable = false)
    private String titre;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String corps;

    @Column(nullable = false)
    private String type;

    @ElementCollection
    @CollectionTable(name = "notification_data", joinColumns = @JoinColumn(name = "notif_id"))
    @MapKeyColumn(name = "data_key")
    @Column(name = "data_value")
    @Builder.Default
    private Map<String, String> data = new HashMap<>();

    @Column(nullable = false)
    @Builder.Default
    private Boolean lue = false;

    @CreatedDate
    @Column(name = "envoye_at", nullable = false, updatable = false)
    private LocalDateTime envoyeAt;

    // Transient getters for Flutter model compatibility
    public boolean isRead() {
        return lue != null && lue;
    }

    public boolean isBookingConfirmed() {
        return "booking_confirmed".equals(type);
    }

    public boolean isEventReminder() {
        return "event_reminder".equals(type);
    }

    public boolean isTicketReady() {
        return "ticket_ready".equals(type);
    }

    public boolean isCancellation() {
        return "cancellation".equals(type);
    }

    public boolean isPromotion() {
        return "promotion".equals(type);
    }
}